### ISSUE TITLE
Improve Developer Experience with better Config Traits

### ISSUE THREAD

**Issue by shawntabrizi** [paritytech/substrate#10687]
There are common types defined within the config of different pallets, such as:

```
type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;

type Call: IsType<<Self as frame_system::Config>::Call>
	+ Dispatchable<Origin = Self::Origin, PostInfo = PostDispatchInfo>
	+ GetDispatchInfo
	+ FullCodec
	+ From<frame_system::Call<Self>>;

type Origin: From<RawOrigin<Self::AccountId, I>>;

etc...
```

All of these types have some complicated traits which are expected from the `construct_runtime!` macro, and often the user struggles to define all those types when doing runtime development.

We could simply wrap all of these traits in a simple to understand trait like:

```
trait FrameCall = IsType<<Self as frame_system::Config>::Call>
	+ Dispatchable<Origin = Self::Origin, PostInfo = PostDispatchInfo>
	+ GetDispatchInfo
	+ FullCodec
	+ From<frame_system::Call<Self>>;
```

Then simply define:

```
type Call: FrameCall
```

or something like that.

**Comment by dharjeezy** [comment]
Hello @shawntabrizi is this particular issue available for pick up?
I am willing to pick it up

**Comment by bkchr** [comment]
> All of these types have some complicated traits which are expected from the `construct_runtime!` macro

That is not correct. You add the traits bounds that you are require for a certain type in your pallet. The runtime is only required to provide types that implement the given traits.

**Comment by shawntabrizi** [comment]
@bkchr I think you are saying what I am saying? 

For example, we know `construct_runtime!` implements a bunch of traits, like `From`/`Into` for inner and outer events, inner and outer origin, inner and outer Call type.

We also know that something like the `Call` type must satisfy a bunch of traits which are expected in other parts of the codebase, like `Dispatchable` or `GetDispatchInfo`. The whole thing I am saying is we can take all these assumptions about common types in the FRAME runtime and place them under a single simple trait.

Or what am I misunderstanding?

**Comment by bkchr** [comment]
That is right what you are saying. I just wanted to highlight that some pallets may only require that a `Call` implements `Encode`. Then you also just need to bound for `Encode`.

**Comment by kianenigma** [comment]
Related to #13454
